backend/test/private-api/private-api.api-tokens.e2e-spec.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * SPDX-FileCopyrightText: 2025 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { ApiTokenWithSecretInterface } from '@hedgedoc/commons';
+import request from 'supertest';
+
+import { PRIVATE_API_PREFIX } from '../../src/app.module';
+import {
+  dateTimeToISOString,
+  getCurrentTimestamp,
+} from '../../src/utils/datetime';
+import { TestSetup, TestSetupBuilder } from '../test-setup';
+import { setupAgent } from './utils/setup-agent';
+
+describe('Tokens', () => {
+  let testSetup: TestSetup;
+
+  let agentNotLoggedIn: request.SuperAgentTest;
+  let agentGuestUser: request.SuperAgentTest;
+  let agentUser1: request.SuperAgentTest;
+  let agentUser2: request.SuperAgentTest;
+
+  beforeEach(async () => {
+    testSetup = await TestSetupBuilder.create().withUsers().build();
+    await testSetup.app.init();
+
+    [agentNotLoggedIn, agentGuestUser, agentUser1, agentUser2] =
+      await setupAgent(testSetup);
+  });
+
+  afterEach(async () => {
+    await testSetup.app.close();
+    await testSetup.cleanup();
+  });
+
+  describe(`POST ${PRIVATE_API_PREFIX}/tokens`, () => {
+    const tokenName = 'private-api-test-token';
+    describe('user can create api tokens', () => {
+      let apiTokenWithSecret: ApiTokenWithSecretInterface;
+      afterEach(() => {
+        expect(apiTokenWithSecret.label).toBe(tokenName);
+        expect(
+          new Date(apiTokenWithSecret.validUntil).getTime(),
+        ).toBeGreaterThan(Date.now());
```

Comment:
Do we want to test that a given validUntil is really respected? Currently we don't do this.